File: Selenium/Activities/Activity7.py
from selenium import webdriver
from selenium.webbrowser.common.by import By

with webbrowser.Firefox() as browser:
    browser.get("https://training-support.net/webelements/dynamic-controls")
    print(f"Current page title: {browser.title}")
	
    text_box = browser.find_element(By.XPATH,"//input[@id = 'textInput']")
    enable_button =  browser.find_element(By.XPATH,"//button[text() = 'Enable Input']")
    print(f"Text box currently visible: {text_box.is_displayed()}")
    print(f"Text box enabled state: {text_box.is_enabled()}")
    enable_button.click()
    print(f"Text box enabled state: {text_box.is_enabled()}")
    browser.quit()

File: Selenium/Activities/Activity7.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity7 {

	public static void main(String[] args) {

			WebDriver browser = new FirefoxDriver();
			browser.get("https://training-support.net/webelements/dynamic-controls");
			System.out.println("Current page title: "+ browser.getTitle());
			
			WebElement textBox = browser.findElement(By.xpath("//input[@id = 'textInput']"));
			WebElement enableButton  =  browser.findElement(By.xpath("//button[text() = 'Enable Input']"));
			System.out.println("Text box currently visible: " + textBox.isDisplayed());
			
			System.out.println("Text box enabled state: " +textBox.isEnabled());
			
			enableButton.click();
				
			System.out.println("Text box enabled state: " +textBox.isEnabled());
			
			browser.quit();

	}

}
